mixed/flink/flink-common/src/main/java/com/netease/arctic/flink/catalog/factories/ArcticCatalogFactory.java

```diff
@@ -57,13 +60,16 @@ public Catalog createCatalog(Context context) {
     final String defaultDatabase = helper.getOptions().get(DEFAULT_DATABASE);
     String metastoreUrl = helper.getOptions().get(METASTORE_URL);
     final Map<String, String> arcticCatalogProperties = getKafkaParams(context.getOptions());
+    final Map<String, String> catalogProperties = Maps.newHashMap(arcticCatalogProperties);
+
+    Optional<String> tableFormatsOptional = helper.getOptions().getOptional(FLINK_TABLE_FORMATS);
+    tableFormatsOptional.ifPresent(
+        tableFormats -> catalogProperties.put(FLINK_TABLE_FORMATS.key(), tableFormats));
 
     return new ArcticCatalog(
```

Comment:
Maybe we should move the class to `factories.mixed` ?

Reply:
makes sense to me.